spark/src/test/scala/org/apache/comet/CometCastSuite.scala

```diff
@@ -895,6 +895,18 @@ class CometCastSuite extends CometTestBase with AdaptiveSparkPlanHelper {
     }
   }
 
+  test("cast between decimals with different precision and scale") {
```

Comment:
It's great to see that some basic tests now pass. I assume there must have been improvements in DataFusion since we started this project.

I'd like to see the tests cover more scenarios, such as:

- Casting from smaller scale to larger scale e.g. (10, 1) to (10, 4)
- Tests for edge cases such as negative scale and zero scale

Reply:
+1 for more scenarios

- edge case inputs like null
- negative scale may not be allowed in Spark IIRC

Reply:
I’m adding more cases, negative scale throws scale exception in spark as well.